astropy/utils/masked/tests/test_function_helpers.py

```diff
@@ -1299,9 +1299,11 @@ def test_array2string(self):
         out2 = np.array2string(self.ma, separator=", ", formatter={"all": hex})
         assert out2 == "[———, 0x1, 0x2]"
         # Also as positional argument (no, nobody will do this!)
-        out3 = np.array2string(
-            self.ma, None, None, None, ", ", "", np._NoValue, {"int": hex}
-        )
+        if NUMPY_LT_2_4:
+            args = (self.ma, None, None, None, ", ", "", np._NoValue, {"int": hex})
+        else:
+            args = (self.ma, None, None, None, ", ", "", {"int": hex})
+        out3 = np.array2string(*args)
```

Comment:
Maybe this part of the test should actually be removed: it seems redundant with in depth signature introspection tests, and it clearly has a maintenance cost of its own. Anyway, that's merely a suggestion for a follow up, but I think it's out of scope here.